src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4259,12 +4265,112 @@ private static BoundExpression SkipReferenceConversions(BoundExpression possibly
             }
         }
 
-        public override BoundNode? VisitConditionalAccess(BoundConditionalAccess node)
+        // PROTOTYPE(improved-definite-assignment):
+        // this is not used outside 'VisitPossibleConditionalAccess' yet, but is expected to be used
+        // when refactoring 'VisitBinaryOperatorChildren'
+        private bool TryVisitConditionalAccess(BoundExpression node, out PossiblyConditionalState stateWhenNotNull)
+        {
+            var (conversion, access) = node switch
+            {
+                BoundConditionalAccess ca => (null, ca),
+                BoundConversion { Conversion: Conversion innerConversion, Operand: BoundConditionalAccess ca } conv when isAcceptableConversion(ca, innerConversion) => (conv, ca),
```

Comment:
I'm don't believe this is good enough. Consider using the `RemoveConversion` helper method, and test scenarios that involve conversion groups (I expect that something like `int? i = (int?)longValue;` should have one).

Reply:
Thanks, I think this is related to the issues in tests `NullCoalescing_CondAccess_ExplicitUserDefinedConv_01_02` and `NullCoalescing_CondAccess_ExplicitUserDefinedConv_05_02`. I'll make an attempt to refactor using RemoveConversion.

Reply:
Done. Also realized there were more bugs in here--tried to arrive at a reasonable design and comment it well enough to be comprehensible.